src/libraries/Microsoft.PowerFx.Core/Public/Types/UnsupportedType.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx.Types
+{
+    /// <summary>
+    /// Type not currently supported by the API.
+    /// </summary>
+    public class UnsupportedType : FormulaType
+    {
+        internal UnsupportedType(DType type)
+            : base(type)
+        {
+        }
+
+        /// <inheritdoc />
+        public override void Visit(ITypeVistor visitor)
+        {
+            visitor.Visit(this);
+        }
+
+        internal string Description => $"kind={_type.Kind} fullTypeName={_type.GetType().FullName}";

```

Comment:
>Description [](http://example.com/codeflow?start=24&length=11)

nit: is this just for debugging? If so, you could use [DebuggerDisplay]

Reply:
Done, please do check.